File: Python/Program/Monster.py
from Character import Character

class Monster(Character):
    def __init__(self, name: str, lives_at: str, main_skill: str, HP: int, item_drops: list, damage_effect: str):
        super().__init__(name, lives_at, main_skill)
        self._HP = HP
        self._item_drops = item_drops
        self._damage_effect = damage_effect

    def get_HP(self) -> int:
        return self._HP

    def set_HP(self, HP: int):
        self._HP = HP

    def get_item_drops(self) -> list:
        return self._item_drops

    def set_item_drops(self, item_drops: list):
        self._item_drops = item_drops
        
    def get_damage_effect(self) -> str:
        return self._damage_effect

    def set_damage_effect(self, damage_effect: str):
        self._damage_effect = damage_effect

File: C++/Program/Monster.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Character.cpp"

using namespace std;

class Monster : public Character {
private:
    int HP;
    vector<string> itemDrops;
    string damageEffect;

public:
    // Constructor
    Monster(){}

    Monster(string name, string livesAt, string mainSkill, int HP, vector<string> itemDrops, string damageEffect)
        : Character(name, livesAt, mainSkill) {
        this->HP = HP;
        this->itemDrops = itemDrops;
        this->damageEffect = damageEffect;
    }

    // Getter and Setter for HP
    int getHP() {
        return HP;
    }

    void setHP(int HP) {
        this->HP = HP;
    }

    // Getter and Setter for itemDrops
    vector<string> getItemDrops() {
        return itemDrops;
    }

    void setItemDrops(vector<string> itemDrops) {
        this->itemDrops = itemDrops;
    }

    // Getter and Setter for damageEffect
    string getDamageEffect() {
        return damageEffect;
    }

    void setDamageEffect(string damageEffect) {
        this->damageEffect = damageEffect;
    }
};